src/main/kotlin/org/rust/ide/intentions/MatchToIfLetIntention.kt

```diff
@@ -33,6 +33,7 @@ class MatchToIfLetIntention : RsElementBaseIntentionAction<MatchToIfLetIntention
             .filter { it.expr?.isVoid == false }
             .singleOrNull() ?: return null
 
+        if (nonVoidArm.matchArmGuard != null) return null
```

Comment:
Maybe we should also check `outerAttrList` for emptiness?

Reply:
Thanks!